build error

```
/root/wasmer/wasmer-c-api/wasmer-c-api-example.c:8:8: warning: type defaults to ‘int’ in declaration of ‘print_str_called’ [-Wimplicit-int]
 static print_str_called = false;
        ^~~~~~~~~~~~~~~~
/root/wasmer/wasmer-c-api/wasmer-c-api-example.c: In function ‘print_str’:
/root/wasmer/wasmer-c-api/wasmer-c-api-example.c:14:31: warning: initialization discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
     wasmer_memory_t *memory = wasmer_instance_context_memory(ctx, 0);
                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/root/wasmer/wasmer-c-api/wasmer-c-api-example.c: In function ‘main’:
/root/wasmer/wasmer-c-api/wasmer-c-api-example.c:36:57: warning: passing argument 1 of ‘wasmer_import_func_new’ from incompatible pointer type [-Wincompatible-pointer-types]
     wasmer_import_func_t *func = wasmer_import_func_new(print_str, params_sig, 2, returns_sig, 0);
                                                         ^~~~~~~~~
In file included from /root/wasmer/wasmer-c-api/wasmer-c-api-example.c:4:0:
/root/wasmer/wasmer-c-api/rust-build/src/wasmer-runtime-c-api/lib/runtime-c-api/wasmer.h:352:23: note: expected ‘void (*)(void *)’ but argument is of type ‘void (*)(wasmer_instance_context_t *, int32_t,  int32_t) {aka void (*)(struct <anonymous> *, int,  int)}’
 wasmer_import_func_t *wasmer_import_func_new(void (*func)(void *data),
                       ^~~~~~~~~~~~~~~~~~~~~~
make[2]: *** No rule to make target 'rust-build/src/wasmer-runtime-c-api/target/debug/libwasmer_runtime_c_api.dylib', needed by 'wasmer-c-api-example'.  Stop.
CMakeFiles/Makefile2:67: recipe for target 'CMakeFiles/wasmer-c-api-example.dir/all' failed
make[1]: *** [CMakeFiles/wasmer-c-api-example.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```    


seems the critical error is:  
```
/root/wasmer/wasmer-c-api/wasmer-c-api-example.c: In function ‘main’:
/root/wasmer/wasmer-c-api/wasmer-c-api-example.c:36:57: warning: passing argument 1 of ‘wasmer_import_func_new’ from incompatible pointer type [-Wincompatible-pointer-types]
     wasmer_import_func_t *func = wasmer_import_func_new(print_str, params_sig, 2, returns_sig, 0);
                                                         ^~~~~~~~~
In file included from /root/wasmer/wasmer-c-api/wasmer-c-api-example.c:4:0:
/root/wasmer/wasmer-c-api/rust-build/src/wasmer-runtime-c-api/lib/runtime-c-api/wasmer.h:352:23: note: expected ‘void (*)(void *)’ but argument is of type ‘void (*)(wasmer_instance_context_t *, int32_t,  int32_t) {aka void (*)(struct <anonymous> *, int,  int)}’
 wasmer_import_func_t *wasmer_import_func_new(void (*func)(void *data),
```

--- a/wasmer-c-api-example.c
+++ b/wasmer-c-api-example.c
@@ -2,15 +2,15 @@
 #include "rust-build/src/wasmer-runtime-c-api/lib/runtime-c-api/wasmer.h"
 #include <assert.h>
 #include <stdint.h>
+#include <string.h>
 
-static print_str_called = false;
+static int print_str_called = false;
 
 // Host function that will be imported into the Web Assembly Instance
 void print_str(wasmer_instance_context_t *ctx, int32_t ptr, int32_t len)
 {
     print_str_called = true;
-    wasmer_memory_t *memory = wasmer_instance_context_memory(ctx, 0);
-    uint32_t mem_len = wasmer_memory_length(memory);
+    const wasmer_memory_t *memory = wasmer_instance_context_memory(ctx, 0);
     uint8_t *mem_bytes = wasmer_memory_data(memory);
     printf("%.*s", len, mem_bytes + ptr);
 }
@@ -31,19 +31,19 @@ int main()
     // of our `print_str` host function
     wasmer_value_tag params_sig[] = {WASM_I32, WASM_I32};
     wasmer_value_tag returns_sig[] = {};
-    wasmer_import_func_t *func = wasmer_import_func_new(print_str, params_sig, 2, returns_sig, 0);
+    wasmer_import_func_t *func = wasmer_import_func_new((void (*)(void *)) print_str, params_sig, 2, returns_sig, 0);
 
     // Create module name for our imports
     // represented in bytes for UTF-8 compatability
-    char *module_name = "env";
+    const char *module_name = "env";
     wasmer_byte_array module_name_bytes;
-    module_name_bytes.bytes = module_name;
+    module_name_bytes.bytes = (const uint8_t *) module_name;
     module_name_bytes.bytes_len = strlen(module_name);
 
     // Define a function import
-    char *import_name = "_print_str";
+    const char *import_name = "_print_str";
     wasmer_byte_array import_name_bytes;
-    import_name_bytes.bytes = import_name;
+    import_name_bytes.bytes = (const uint8_t *) import_name;
     import_name_bytes.bytes_len = strlen(import_name);
     wasmer_import_t func_import;
     func_import.module_name = module_name_bytes;
@@ -52,9 +52,9 @@ int main()
     func_import.value.func = func;
 
     // Define a memory import
-    char *import_memory_name = "memory";
+    const char *import_memory_name = "memory";
     wasmer_byte_array import_memory_name_bytes;
-    import_memory_name_bytes.bytes = import_memory_name;
+    import_memory_name_bytes.bytes = (const uint8_t *) import_memory_name;
     import_memory_name_bytes.bytes_len = strlen(import_memory_name);
     wasmer_import_t memory_import;
     memory_import.module_name = module_name_bytes;
@@ -75,9 +75,9 @@ int main()
     memory_import.value.memory = memory;
 
     // Define a global import
-    char *import_global_name = "__memory_base";
+    const char *import_global_name = "__memory_base";
     wasmer_byte_array import_global_name_bytes;
-    import_global_name_bytes.bytes = import_global_name;
+    import_global_name_bytes.bytes = (const uint8_t *) import_global_name;
     import_global_name_bytes.bytes_len = strlen(import_global_name);
     wasmer_import_t global_import;
     global_import.module_name = module_name_bytes;
@@ -105,10 +105,12 @@ int main()
     wasmer_instance_t *instance = NULL;
     wasmer_result_t compile_result = wasmer_instantiate(&instance, bytes, len, imports, 3);
     printf("Compile result:  %d\n", compile_result);
+
     if (compile_result != WASMER_OK)
     {
         print_wasmer_error();
     }
+
     assert(compile_result == WASMER_OK);
 
     // Call the exported "hello_wasm" function of our instance

